controllers/factory/static_targets_build.go

```diff
@@ -66,6 +68,9 @@ func generateStaticScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "params", Value: ep.Params})
 	}
 	if ep.Scheme != "" {
```

Comment:
I think about adding validation at `UnmarshalJSON` method. `Scheme` could be wrapped into custom type.
 It allows to validate it with `AdmissionWebhook` and fail early. 

 It's already done for alertmanager config:
https://github.com/VictoriaMetrics/operator/blob/master/api/v1beta1/vmalertmanagerconfig_types.go#L193

Reply:
Updated, using kubebuilder CRD validation markers now.